### ISSUE TITLE
[sl-form] Error messages should use sl-announcer

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3680]
### 😯 Current Behavior

When screen reader user validate form field either using on-submit or on-blur validation the error message is not automatically announced by SR without going back to invalid field.

That makes filling the form more difficult, as user has to go back to each field in order to check if there are any errors.

### 🤔 Expected Behavior/possible solution?

`sl-error` content should be automatically announced by screen readers when error message gets displayed in form.

Probably the easiest way for that is to use `sl-announcer` and add there all the error messages when their appear? This should work smoothly for small number of errors but should be also tested on [Form - All](https://storybook.sanomalearning.design/?path=/story/form-form--all) when all the errors are submitted at once. Expected behavior is that user gets to hear all the error messages at once if keyboard focus stays on submit button.

### 🖥️ Steps to Reproduce

1. With screen reader turned on go to: https://storybook.sanomalearning.design/?path=/story/form-form--all
2. Submit empty form
3. Verify if screen reader automatically reads all the error messages without moving keyboard focus from submit button.

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-form--all

### ⚖️ Impact

This blocks our release or stops us from using this component

### Framework

Lit

### What browsers are you seeing the problem on?

All

### Component version

1.4.2

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by Diaan** [comment]
Right now all error message contain "this field" which makes reading the standard sl-error message not very useful. 
What happens when you have multiple errors?

For now we add the announcer to sl-error so Kacper can see what happens when it IS announced.

**Comment by a11ymiko** [comment]
There should be two ways of announcing errors in the form with `on submit` validation. Each form should use only one option for whole form:
1. **Inline message with summary of all errors.** Inline message content is automatically announced by `sl-announcer` so screen reader user will get necessary information about where and what errors were found. In this scenerio each `sl-error` should not be announced by`sl-announcer` as it would make error announcement unnecessary longer and harder to understand. When there is inline messsage with error summary only the inline message should be announced automatically by SRs.

<img width="1189" height="176" alt="Image" src="https://github.com/user-attachments/assets/b50da753-d971-4430-bdb9-03344cfafa5b" />

2. **Automatically announced error messages for individual fields.** If there is no inline message with summary of errors, then all the error messages that are displayed below each field should be announced by `sl-announcer`. This announcement should consist of **label of invalid field** AND **its error message**. This way screen reader user will automatically get information about what fields needs to be fixed and how. It's crucial in this scenerio that all the errors are announced by screen reader so content of all errors should be added to `sl-announcer` at once, as one content. Adding this errors one by one can cause screen reader to skip some announcements in the middle of the sentence to start new announcement because of delay issues.

<img width="1199" height="371" alt="Image" src="https://github.com/user-attachments/assets/02968ec3-c648-4a42-b3a9-d25b21e2e8b3" />

For `on blur` validation when error is displayed on focus change then it should automatically announced by screen reader. It's then recommended to add higher priority to the announcement (so `aria-live='assertive'` or `role='alert'` instead of `aria-live='polite'`) so the error is announced before screen reader start to read the next focused form field. Then it's less likely for user to miss error information.
